### ISSUE TITLE
why length=len(self.train_loader) other than len(self.train_loader.dataset）?

### ISSUE THREAD

**Issue by pangda72** [CN-zdy/MSU_Net#17]
Hi CN-zdy
Thank you for your code . I have some queries regarding your work? Colud you please give me a hint,thank you very much!

In the "solver.py"
why length=len(self.train_loader) other than length=len(self.train_loader.dataset)?

len(trian_loader) is equal to the batch num(batch num=total number of images in training set / batchsize)
But  length=len(self.train_loader.dataset) is equal to total number of images in training set

In this [link](https://github.com/LeeJunHyun/Image_Segmentation/blob/master/solver.py):length=length+images.size(0),is equal to total number of images in training set,I don't know which one is right, Could you tell me?

Thanks in advance.
Looking forward to your reply  .
Best.

**Comment by CN-zdy** [comment]
Hi, Pangda.
LeeJunHyun calculated the results of each image, while I calculated the results of each batch.

**Comment by pangda72** [comment]
> Hi, Pangda.
> LeeJunHyun calculated the results of each image, while I calculated the results of each batch.

Hi CN-zdy
Thank you for your guidance.Helped me solve a problem that had been bothering me.Thank you so much indeed.
Best wishes.

**Comment by pangda72** [comment]
> Hi, Pangda.
> LeeJunHyun calculated the results of each image, while I calculated the results of each batch.

Hi CN-zdy
Thank you for your guidance and help me figure out the above problems.But then I had another problem.I don't know whether the average accuracy、 average dice.... is calculated by reading batchsize images each time or overall accuracy、overall dice.....Colud you  give me a Help?
Thanks in advance.
Best!
